README.md

```diff
@@ -0,0 +1,536 @@
+    {:ok, pid} = :emqtt.start_link(emqtt_opts)
+
+    timer = :erlang.start_timer(interval, self(), :tick)
+    {:ok,
+      %{
+        interval: interval,
+        timer: timer,
+        report_topic: report_topic,
+        pid: pid
+      },
+      {:continue, :start_emqtt}}
+  end
+
+  def handle_continue(:start_emqtt, %{pid: pid} = st) do
+    # Here we make actual connection to the broker
+    {:ok, _} = :emqtt.connect(pid)
+    emqtt_opts = Application.get_env(:weather_sensor, :emqtt)
+    clientid = emqtt_opts[:clientid]
+    # From this topic we will receive commands
+    {:ok, _, _} = :emqtt.subscribe(pid, {"/commands/#{clientid}/set_interval", 1})
+    {:noreply, st}
+  end
+
+  def handle_info({:timeout, _ref, :tick}, %{report_topic: topic, pid: pid} = st) do
+    report_temperature(pid, topic)
+    {:noreply, ensure_timer(st)}
+  end
+
+  def handle_info({:publish, publish}, st) do
+    IO.inspect(publish)
+    handle_publish(parse_topic(publish), publish, st)
+  end
+
+  defp handle_publish(["commands", _, "set_interval"], %{payload: payload}, st) do
+    new_interval = String.to_integer(payload)
+    {:noreply, %{st | interval: new_interval}}
+  end
+
+  defp handle_publish(_, _, st) do
+    {:noreply, st}
+  end
+
+  defp ensure_timer(%{timer: timer, interval: interval} = st) do
+    :erlang.cancel_timer(timer)
+    new_timer = :erlang.start_timer(interval, self(), :tick)
+    %{st | timer: new_timer}
+  end
+
+  defp parse_topic(%{topic: topic}) do
+    String.split(topic, "/", trim: true)
+  end
+
+  defp report_temperature(pid, topic) do
+    temperature = 10.0 + 2.0 * :rand.normal()
+    message = {:os.system_time(:millisecond), temperature}
+    payload = :erlang.term_to_binary(message)
+    :emqtt.publish(pid, topic, payload)
+  end
+end
+```
+
+And put some options to `config/config.exs`:
+
+```elixir
+import Config
+
+config :weather_sensor, :emqtt,
+  host: '127.0.0.1',
+  port: 1883,
+  clientid: "weather_sensor",
+  clean_start: false,
+  name: :emqtt
+
+config :weather_sensor, :interval, 1000
+```
+
+Let us summarize a bit what happens in `WeatherSensor`:
+* It implements `GenServer` behaviour.
+* When starting, it
+    * opens an MQTT connection;
+    * subscribes to `/commands/weather_sensor/set_interval` topic for receiving commands, received data will
+    be sent to the process by `:emqtt` as `{:publish, publish}` messages.
+    * schedules timer with a predefined interval.
+* On timer timeout, it publishes `{Timestamp, Temperature}` tuple to `/reports/weather_sensor/temperature` topic.
+* On receiving a message from `/commands/weather_sensor/set_interval` topic, it updates timer interval.
+
+Since our application is not a real Nerves application with a sensor like BMP280 attached, we generate temperature data.
+
+Here we can already see one advantage over HTTP interaction: we can not only send data,
+but also receive some commands in real-time.
+
```

Comment:
Suggestion: warn the eager reader that EMQX must be running before they start the application at this point, else it'll fail to start because it can't connect to the broker. :wink: